File: project/sdk/android/NexusChat.kt
package com.nexus.chat

import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONObject

/** Minimal Nexus chat client for Android. */
class NexusChat(
    private val apiBase: String,
    private var token: String,
    private val sessionId: String = "android-${System.currentTimeMillis()}",
) {
    private val client = OkHttpClient()
    private val json = "application/json; charset=utf-8".toMediaType()

    fun send(message: String): String {
        val body = JSONObject()
            .put("message", message)
            .put("session_id", sessionId)
            .put("agent_id", "chat_support")
            .toString()
            .toRequestBody(json)
        val req = Request.Builder()
            .url("$apiBase/chat")
            .addHeader("Authorization", "Bearer $token")
            .post(body)
            .build()
        client.newCall(req).execute().use { resp ->
            val obj = JSONObject(resp.body?.string() ?: "{}")
            return obj.optString("response", obj.optString("detail", ""))
        }
    }
}

File: project/sdk/ios/NexusChat.swift
import Foundation

/// Minimal Nexus chat client for iOS — wraps REST /chat API.
public final class NexusChat {
    public let apiBase: URL
    public var token: String
    public var sessionId: String

    public init(apiBase: URL, token: String, sessionId: String = UUID().uuidString) {
        self.apiBase = apiBase
        self.token = token
        self.sessionId = sessionId
    }

    public func send(message: String) async throws -> String {
        var req = URLRequest(url: apiBase.appendingPathComponent("chat"))
        req.httpMethod = "POST"
        req.setValue("application/json", forHTTPHeaderField: "Content-Type")
        req.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
        let body: [String: String] = [
            "message": message,
            "session_id": sessionId,
            "agent_id": "chat_support",
        ]
        req.httpBody = try JSONSerialization.data(withJSONObject: body)
        let (data, _) = try await URLSession.shared.data(for: req)
        let json = try JSONSerialization.jsonObject(with: data) as? [String: Any]
        return json?["response"] as? String ?? ""
    }
}
